### PR TITLE
Upgrade Rails from '5.2.7.1' to '6.0.6.1'

### PR DESCRIPTION
**Pull request by Drew-Goddyn**
- Upgrades rails to 6.0.6.1
- Upgrades other gems in Gemfile through `bundle update`
- Ran `app:update` to use new rails configuration defaults
- Ran `rubocop -a` to auto correct linting issues based on enforced rules

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -29,22 +29,17 @@
...
   # Enable serving of images, stylesheets, and JavaScripts from an asset server.
-  # config.action_controller.asset_host = "http://assets.example.com"
+  # config.action_controller.asset_host = 'http://assets.example.com'
 
   # Specifies the header that your server uses for sending files.
-  # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for Apache
-  # config.action_dispatch.x_sendfile_header = "X-Accel-Redirect" # for NGINX
-
-  # Store uploaded files on the local file system (see config/storage.yml for options)
-  config.active_storage.service = :local
```
Comment (anthonyoconnorclio) [inline config/environments/production.rb:42]: is this removed on purpose? Or is it no longer a valid setting?

File: config/environments/production.rb [hunk 2]
```diff
@@ -80,15 +75,33 @@
...
+  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
 
   if ENV["RAILS_LOG_TO_STDOUT"].present?
     logger           = ActiveSupport::Logger.new(STDOUT)
     logger.formatter = config.log_formatter
     config.logger    = ActiveSupport::TaggedLogging.new(logger)
   end
 
-  # Do not dump schema after migrations.
-  config.active_record.dump_schema_after_migration = false
```
Comment (anthonyoconnorclio) [inline config/environments/production.rb:93]: was this removed because the default is false?

Comment (Drew-Goddyn) [reply config/environments/production.rb:93]: the default doesn't even include that line anymore, and this repo has no database at all so it felt unncessary.

File: config/environments/production.rb (hunk 1 above)
Comment (peerkleio) [reply config/environments/production.rb:42]: We don't use active_storage so I suspect its unnecessary.

**Review by peerkleio — APPROVED** [review]

File: config/environments/production.rb (hunk 1 above)
Comment (Drew-Goddyn) [reply config/environments/production.rb:42]: I believe the default from app:update removed it but we also don't use active storage and have the railtie commented out [here](https://github.com/clio/example-third-party-application/pull/31/files#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9R8)

**Review by anthonyoconnorclio — APPROVED** [review]

**Pull request merged**